src/main/java/org/eclipse/cargotracker/interfaces/booking/facade/dto/CargoTracking.java

```diff
@@ -0,0 +1,113 @@
+package org.eclipse.cargotracker.interfaces.booking.facade.dto;
+
+import org.eclipse.cargotracker.domain.model.cargo.Cargo;
+import org.eclipse.cargotracker.domain.model.cargo.Delivery;
+import org.eclipse.cargotracker.domain.model.cargo.HandlingActivity;
+import org.eclipse.cargotracker.domain.model.handling.HandlingEvent;
+import org.eclipse.cargotracker.domain.model.location.Location;
+
+import java.text.SimpleDateFormat;
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.Date;
+import java.util.List;
+
+public class CargoTracking {
+    private static final SimpleDateFormat DATE_FORMAT = new SimpleDateFormat("MM/dd/yyyy hh:mm a z");
+
+    private final Cargo cargo;
+    private final List<HandlingTrackingEvents> events;
+
+    public CargoTracking(Cargo cargo, List<HandlingTrackingEvents> handlingEvents) {
+        this.cargo = cargo;
```

Comment:
This still looks like a view adapter, not a DTO. The assembler should completely copy the domain object and keep the DTO very simple. For a good example, please look at CargoRouteDtoAssembler?

Reply:
Was not sure on that part, thanks to point this, I'll fix.